### PR TITLE
price-protection fixes and tests

### LINKED ISSUE #6001: 1870: Share Count Incorrectly Calculated during price protect
https://18xx.games/game/40469?action=883

At this point in the game, it says that Roller has 9/16 shares.

They hold:

SP (60%*) (0 - in the yellow area)
FW*	50% (4)
ATSF*50% (4)
MKT*40% (3)
SLSF 10% (1)
GMO 10% (1)

This adds up to 13. When they price protect 4 shares, their share count jumps up to 17/16, which is correct but shouldn't be allowed as price protection isn't an option if it will push you over share limit.

### PR DESCRIPTION
**Pull request by JoeB989**
Fixed #6001
Fixed cert counting of white/yellow shares during price protection 
Added in-game test spec and price-protect fixtures

### CODE REVIEW

File: spec/lib/engine/games/game_spec.rb [hunk 1]
```diff
@@ -106,4 +106,31 @@ module Engine
       expect(game.round.active_step).to be_instance_of(Engine::Step::Route)
     end
   end
+
+  # active-game tests: run all .ingame files found in spec/fixtures/,
+  # verify that the result and last "test_last_actions" actions match those in the JSON
+  Find.find(FIXTURES_DIR).select { |f| File.basename(f) =~ /.ingame/ }.each do |fixture|
```
Comment (tobymao) [inline spec/lib/engine/games/game_spec.rb:112]: can you just do the normal one and delete this method?

Comment (JoeB989) [reply spec/lib/engine/games/game_spec.rb:112]: The normal one expects completed game, this one expects partial game stopped where the actions to be tested are.  I could remove the` expect(rungame.finished).to eq(true)` above, but I didn't know what other side effects I might run into.

Comment (tobymao) [reply spec/lib/engine/games/game_spec.rb:112]: can you just end your games with a end game action?

Comment (JoeB989) [reply spec/lib/engine/games/game_spec.rb:112]: ah hopefully.  I 'll try

**Review by tobymao — APPROVED** [review]

**Pull request merged**